### PR TITLE
update_icon for atoms

### PR DESCRIPTION
**Pull request by AlexMorgan3817**
## About The Pull Request
Created atom proc update_icon() and refactored bad code formating in worm.dm <details>Ignore diff, he writes some crap in worm.dm</details>

### CODE REVIEW

File: code/modules/mob/living/simple_animal/worm.dm [hunk 1]
```diff
@@ -44,154 +44,154 @@
...
-		update_icon()
+/mob/living/simple_animal/space_worm/Life()
+	..()
 
-		return
+	if(next && !(next in view(src,1)))
+		Detach()
 
-	Destroy() //if a chunk a destroyed, make a new worm out of the split halves
+	if(stat == DEAD) //dead chunks fall off and die immediately
```
Comment (TorinoFermic) [inline code/modules/mob/living/simple_animal/worm.dm]: We have macros for this like IsDead()

File: code/modules/mob/living/simple_animal/worm.dm [hunk 2]
```diff
@@ -44,154 +44,154 @@
...
-		if(stat == DEAD) //dead chunks fall off and die immediately
-			if(previous)
-				previous.Detach()
-			if(next)
-				Detach(1)
-
-		if(prob(stomachProcessProbability))
-			ProcessStomach()
+/mob/living/simple_animal/space_worm/head/update_icon()
+	if(stat == CONSCIOUS || stat == UNCONSCIOUS)
```
Comment (TorinoFermic) [inline code/modules/mob/living/simple_animal/worm.dm]: Ditto here

Comment (LethalGhost) [reply code/modules/mob/living/simple_animal/worm.dm]: @TorinoFermic that's bad code formatting but it's valid check. Right now stat can be `CONSCIOUS`, `UNCONSCIOUS` and `DEAD`. Actually that check means `if(stat != DEAD)`

File: code/modules/mob/living/simple_animal/worm.dm (hunk 1 above)
Comment (AlexMorgan3817) [reply code/modules/mob/living/simple_animal/worm.dm]: Isn't my code, it was already here.

File: code/modules/mob/living/simple_animal/worm.dm (hunk 2 above)
Comment (AlexMorgan3817) [reply code/modules/mob/living/simple_animal/worm.dm]: Isn't my code, it was already here.

**Pull request merged**